New Rule: 2.4.1 Bypass Blocks (cf77f2) (#421)
act-rules/act-rules.github.io: _rules/block-collapsible-3e12e1.md

## Test Cases

**Note**: The text of the examples is from the translation of the first Chapter of _The Three Kingdoms_ by Yu Sumei (Tuttle publishing, May 2014).

**Note**: Unless specified otherwise, the [sections of content][] of each document are defined by the [landmarks][landmark] (`aside`, `main` and `nav` elements); both the complementary and navigational [sections of content][] (`aside` and `nav` elements) are [sections of repeated content][section of repeated content].

### Passed

#### Passed Example 1

The [visibility][visible] and [inclusion in the accessibility tree][included in the accessibility tree] of the navigational [section of repeated content][] can be toggled on and off by the link at the start of the document.

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleHidden('navigation')">Toggle table of content</a>
		<nav id="navigation">
			<h1>Contents</h1>
			<!-- list of links to each chapter, repeated on each page -->
		</nav>
		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>

> 			Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span of

Reviewer: Why is this linebreak here in every example?

Reviewer 2: That's HTML prettier cutting lines at 120 characters (iirc).
I agree that it's super ugly in that case, but there is not much we can do about it…

			Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span of
			time.
		</main>
	</body>
</html>
```

### Inapplicable

#### Inapplicable Example 1

This [document][] is not an [HTML web page][].

```svg
<svg xmlns="http://www.w3.org/2000/svg">
  <title>This is an SVG</title>
</svg>
```

[accessible name]: #accessible-name 'Definition of accessible name'
[document]: https://dom.spec.whatwg.org/#concept-document 'Definition of document'
[document element]: https://dom.spec.whatwg.org/#document-element 'Definition of document element'
[included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
[landmark]: https://www.w3.org/TR/wai-aria-1.1/#landmark_roles 'List of landmark roles'
[tech scr28]: https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR28 'Technique SCR28: Using an expandable and collapsible menu to bypass block of content'
[sections of content]: #section-of-content 'Definition of section of content'
[section of repeated content]: #section-of-repeated-content 'Definition of section of repeated content'

> [user interface component]: https://www.w3.org/TR/WCAG21/#dfn-user-interface-components 'Definition of user interface component'

Reviewer 3: This is an ambiguous definition. We can't use this here.

	</head>
	<body>
		<button onclick="toggleHidden('chapters-navigation', 'about-book')">Toggle repeated content</button>

		<nav id="chapters-navigation">
			<ol>
				<li><a>Chapter 1</a></li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<aside id="about-book">
			<p>The Romance of the Three Kingdoms is a 14th century historical novel.</p>
		</aside>

		<div id="main">
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</div>
	</body>
</html>
```

#### Passed Example 3

In this document, the [visibility][visible] and [inclusion in the accessibility tree][included in the accessibility tree] of both the complementary and the navigational [blocks of repeated content][block of repeated content] can be toggled on and off, each of them by a different [instrument][].

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleHidden('chapters-navigation')">Toggle table of content</a>

> 		<button onclick="toggleHidden('bio-translator')">Toggle extra content</button>

Reviewer 3: Looks like it should be this:

```suggestion
		<button onclick="toggleHidden('about-book')">Toggle extra content</button>
```

			<p>The Romance of the Three Kingdoms is a 14th century historical novel.</p>
		</aside>

		<div id="main">
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</div>
	</body>
</html>
```

#### Passed Example 4

In this document, the [visibility][visible] of the navigational [block of repeated content][] can be toggled on and off by the [visible][] link at the start of the document. Its [inclusion in the accessibility tree][included in the accessibility tree] can be toggled on and off by the button at the start of the document. Note that having a [focusable][] [user interface component][] with `aria-hidden` is a violation of both [Success Criterion 1.3.1 Info And Relationships](https://www.w3.org/tr/wcag21/#info-and-relationships) and [4.1.2 Name, Role, Value](https://www.w3.org/tr/wcag21/#name-role-value) and must thus be avoided.

```html
<html>
	<head>
		<link rel="stylesheet" href="../test-assets/bypass-blocks-cf77f2/styles.css" />
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>

> 		<a href="#" onclick="toggleVisibility('chapters-navigation')" aria-hidden="true">Toggle table of content</a>

Reviewer 3: Don't think this should have aria-hidden, since this element is focusable.

Reviewer 2: After much thoughts, I'm dropping the example altogether.

The point was to try and show that different instruments could be used to to toggle visibility and inclusion in the accessibility tree. But then the instrument toggling visibility was not doing anything for screen reader users, hence the `aria-hidden`, which creates this issue, and adding a negative `tabindex` means that sighted keyboard users are lent without a way to toggle the repeated content.

So, I guess it's just better to drop the example…

---
id: 3e12e1
name: Block of content is expandable and collapsible
rule_type: atomic
description: |
  This rule checks that repeated blocks of content are expandable and collapsible
accessibility_requirements:
  wcag-technique:SCR28: # Using an expandable and collapsible menu to bypass block of content
    forConformance: false
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS styling
acknowledgments:
  authors:
    - Jean-Yves Moyen
  assets:
    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
---

## Applicability

This rule applies to any [HTML web page][].

## Expectation 1

> There exists a [semantic segmentation][] of the test target such that for each [block][] in this [segmentation] which is before the [main block of content][] and contains at least one [block of repeated content][], there exists a [visible][] [instrument][] to make this [block][] not [visible][].

Reviewer 2: I'm naming the block to make it clear.

> Also, cannot the block of repeated content be the other block? Does it have to be contained in the other block?

Yes, it can be the same. A set is included in itself 😉 I don't know if I should make that explicit. The examples are illustrating that since the blocks of the segmentation are exactly the blocks of repeated content.

> There exists a [semantic segmentation][] of the test target such that for each [block][] in this [segmentation] which is before the [main block of content][] and contains at least one [block of repeated content][], there exists a [visible][] [instrument][] to make this [block][] not [visible][].

Reviewer 4: That is true. Perhaps the examples are enough.

## Assumptions

- This rule assumes that there is exactly one [main block of content][] inside each [HTML web page][].
- This rule assumes that completely removing [blocks of repeated content][block of repeated content] is sufficient to pass [Success Criterion 2.4.1: Bypass blocks][sc241]. [Technique SCR28: Using an expandable and collapsible menu to bypass block of content][tech scr28] does require that they can be toggled on and off. In any case, providing an [instrument][] to remove them without providing one to show them again is likely going to create other unrelated issues.

## Accessibility Support

_There are no major accessibility support issues known for this rule._

## Background

- [SCR28: Using an expandable and collapsible menu to bypass block of content][tech scr28]

> In the test cases, the `aside` and `nav` elements are each a [block of repeated content][] due to the link inside the `nav` element to a page with similar [blocks of content][block of content]; and the `main` element is the [main block of content][]. Unless specified, the [semantic segmentation][] fulfilling the expectations has one block for the `main` element; one for each of the `aside` and `nav` elements (if present); and one for the other content, if any.

Reviewer: Would it not be "each considered as a"

> In the test cases, the `aside` and `nav` elements are each a [block of repeated content][] due to the link inside the `nav` element to a page with similar [blocks of content][block of content]; and the `main` element is the [main block of content][]. Unless specified, the [semantic segmentation][] fulfilling the expectations has one block for the `main` element; one for each of the `aside` and `nav` elements (if present); and one for the other content, if any.

Reviewer 2: I disagree with this.
In that case, the blocks are the exact same across the pages. So there is no interpretation left to the possibility of being "equivalent resource" or not and it is clear that they **are** blocks of repeated content without the need to consider this possibility.
If the block were containing similar but different information, then I agree that "considered as" would be a good formulation as there is some room for interpretation.

---
id: 3e12e1
name: Block of repeated content is collapsible
rule_type: atomic
description: |
  This rule checks that repeated blocks of content are collapsible
accessibility_requirements:
  wcag-technique:SCR28: # Using an expandable and collapsible menu to bypass block of content
    forConformance: false
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - Accessibility Tree
  - CSS Styling
  - DOM Tree
acknowledgments:
  authors:
    - Jean-Yves Moyen
  assets:
    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
---

## Applicability

This rule applies to any [HTML web page][].

## Expectation

For each [block of repeated content][] in each test target, which is before (in the [flat tree][]) at least one node of [perceivable content][] belonging to no [block of repeated content][], all the following are true:

- there exists a [keyboard actionable][] [instrument][] to make all nodes in this [block][] not [visible][]; and

> - there exists an [instrument][], which is [included in the accessibility tree][], to remove all nodes in this [block][] from the [accessibility tree][included in the accessibility tree].

Reviewer 2: 🤔 
That note is actually part of the WCAG definition of "mechanism" which is used in the SC, so it's not just Understanding material but is part of the normative WCAG document, although only as a note rather than the main text…

---
So, there's one question of whether WCAG's "mechanism" must meet all SC at the same conformance level of the criterion is it used.

I would be at least a bit reluctant in saying that `<span  onclick="location.assign('#main');" aria-hidden="true">Click if you dare</span>` is sufficient to satisfy 2.4.1. It certainly is a mechanism to bypass block of repeated content, but it doesn't add much value…

---
And there's another question of whether we want the rule(s) to require that.

---
I'm seeing the two usual solutions:
- either we drop the requirement from the rule and add an Accessibility Support note saying that the rule is fairly lenient;
- or we keep the requirement in the rule with an assumption that this is part of the definition of "mechanism" (currently having an assumption for keyboard accessibility, would need one for inclusion in the accessibility tree).

I'm slightly more in favour of the second option given that the note on meeting other SC is part of WCAG's definition of "mechanism". But it's not a strong opinion.

---
Other rules that translate WCAG's mechanism (e.g. the single character shortcut one) do not have these requirements. I don't remember if this was discussed during the writing of the rule. But that leads me to another way to handle the question:
1. Locally remove these requirements from the rule. At least allows to ship the rule faster 🤞 
2. Open an Agenda item with the question since it is affecting several rules.
3. Depending on the outcome of the discussion, update our definition of "instrument" which is our translation of WCAG's mechanism (possibly simply adding an Accessibility Support note to it).

---
id: 3e12e1
name: Block of content is expandable and collapsible
rule_type: atomic
description: |
  This rule checks that repeated blocks of content are expandable and collapsible
accessibility_requirements:
  wcag-technique:SCR28: # Using an expandable and collapsible menu to bypass block of content
    forConformance: false
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS styling
acknowledgments:
  authors:
    - Jean-Yves Moyen
  assets:
    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
---

## Applicability

This rule applies to any [HTML web page][].

## Expectation 1

> There exists a [semantic segmentation][] of the test target such that for each [block][], B, in this [segmentation] which is before the [main block of content][] and contains at least one [block of repeated content][], there exists a [visible][] [instrument][] to make this [block][] (B) not [visible][].

Reviewer 5: The SC does not require that the instrument should be visible and in addition, this SC is a related screen reader user who doesn't need any visible instrument as long as it is keyboard accessible such as focusable. I personally conducted interviews with a screenreader user who even does not have a monitor or screen. 

Well, here we need to define criteria if you agree, for example in how many tabs from the start of the page this instrument which must be keyboard accessible when getting focus should be positioned such as within the first three tabs.

Reviewer: This SC is related to more users than just screenreader users. Keyboard users that don't want to tab past a large main menu every time they navigate to a different page would also be helped by an instrument such as this and I would say that if such a mechanism is not visible to such users it is not "available" and thus should fail the SC.

Reviewer 2: More thinking about this, and I tend to agree with you (maybe not for the same reasons).

---
If there is a menu labelled "hide blocks" which, when open, allow to toggle on and off each block, I would say that this satisfy 2.4.1, but it wouldn't pass this rule because the instruments are the items in the menu, hidden by default (the menu itself is not the instrument since its activation only opens it but does not toggle the blocks).

So, it is probably better to require the instrument for toggling visibility to be visible when focused, but not necessarily always.

@ShadowBB, @carlosapaduarte WDYT?

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleHidden('chapters-navigation')">Toggle table of content</a>
		<button onclick="toggleHidden('bio-translator')">Toggle extra content</button>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<aside id="about-book">
			<h1>About the book</h1>
			<p>The Romance of the Three Kingdoms is a 14th century historical novel.</p>
		</aside>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Passed Example 5

> The [visibility][visible] of the navigational [block of repeated content][] can be toggled on and off by the [visible][] link at the start of the document. Its [inclusion in the accessibility tree][included in the accessibility tree] can be toggled on and off by the button at the start of the document.

Reviewer 2: Yes. This is voluntary to point out that there are here two instruments, one for each expectation. I believe that the repetition helps pointing that.

```html
<html>
	<head>
		<link rel="stylesheet" href="../test-assets/bypass-blocks-cf77f2/styles.css" />
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleVisibility('chapters-navigation')" aria-hidden="true">Toggle table of content</a>
		<button onclick="toggleAriaHidden('chapters-navigation')" class="off-screen">Toggle table of content</button>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Passed Example 6

> The navigational [block of repeated content][] can be made non-[visible][] and removed from the accessibility tree by the button at the start of the document.

Reviewer 2: This example shows that hiding forever (no way of showing again) is OK to pass the rule (and the SC, according to the assumption made by the rule).
This is different from other examples that are all using toggling rather than hiding.

Passed Example 1 is the "base case" of what is likely the easiest and best way to pass the rule. Passed Example 2-6 are showing small changes around the expectations that are also OK. Each of them is focused on a single small change in order to not overload reader, and help pinpoint possible issues with implementations.

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<button onclick="hide('chapters-navigation')">Hide table of content</button>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Passed Example 7

> This [document][] has no [block of repeated content][] because there is no other page linked from it.

Reviewer: The applicability is "This rule applies to any [HTML web page][].". This is an HTML page, so it should not be inapplicable.

```html
<html lang="en">
	<head>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<aside id="about-book">
			<h1>About the book</h1>
			<p>The Romance of the Three Kingdoms is a 14th century historical novel.</p>
		</aside>

		<main>
			<h1 id="main">Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

### Failed

#### Failed Example 1

There is no [instrument][] to toggle the navigational [block of repeated content][].

```html
<html>
	<head>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Failed Example 2

> The [instrument][] to toggle the navigational [block of repeated content][] is not [visible][].

Reviewer 5: This example passes the SC and somehow the SCR28 as the instrument is keyboard accessible and hides the navigational content. Consider my comment in the expectation 1 section. Again screen readers don't need any visible instrument to toggle on or off this kind of instrument as long as these are keyboard accessible when getting focus.

Reviewer 2: I disagree that this example passes the SC.
The SC is also beneficial for sigthed users. The Understanding document lists keyboard users, screen magnifier users, and users with cognitive disabilities as benefiting from it.

In this example, a sigthed user doesn't really have a way to bypass the block of repeated content (well, technically it is possible to blindly tab into an out-of-screen link and activate it hoping it does what is needed, but I don't really think this is enough).

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" style="position: absolute; top: -999px" onclick="toggleHidden('chapters-navigation')"
			>Toggle table of content</a
		>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Failed Example 3

> The [instrument][] to toggle the navigational [block of repeated content][] is not [included in the accessibility tree][].

Reviewer 5: This actually works fine that the instrument is not included in the accessibility tree but it passes the expectation 1 and as well as hide the navigational content, which again for sighted use doesn't a problem but for the screen reader user is a big accessibility issue.

Reviewer 2: Here also, I wouldn't consider that AT users have a way of bypassing blocks. Essentially, they can only click on a random link and hope that the best happen.

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" aria-hidden="true" onclick="toggleHidden('chapters-navigation')">Toggle table of content</a>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Failed Example 4

> There is no [instrument][] to toggle the [inclusion in the accessibility tree][included in the accessibility tree] of the navigational [block of repeated content][].

Reviewer 5: ```suggestion
There is [instrument][] to toggle the [inclusion in the accessibility tree][included in the accessibility tree] of the navigational [block of repeated content][] but it does not toggle it.
```

```html
<html>
	<head>
		<link rel="stylesheet" href="../test-assets/bypass-blocks-cf77f2/styles.css" />
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleVisibility('chapters-navigation')">Toggle table of content</a>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

#### Failed Example 5

> There is no [instrument][] to toggle the [visibility][visible] of the navigational [block of repeated content][].

Reviewer 5: ```suggestion
There is [instrument][] to toggle the [visibility][visible] of the navigational [block of repeated content][] but it does not toggle it.
```

```html
<html>
	<head>
		<script src="../test-assets/bypass-blocks-cf77f2/toggle-display.js"></script>
		<title>The Three Kingdoms, Chapter 1</title>
	</head>
	<body>
		<a href="#" onclick="toggleAriaHidden('chapters-navigation')">Toggle table of content</a>

		<nav id="chapters-navigation">
			<h1>Content</h1>
			<ol>
				<li>Chapter 1</li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<main>
			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
			<p>
				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
				of time.
			</p>
		</main>
	</body>
</html>
```

> ### Inapplicable

Reviewer 5: Passed example #7 can be used here?

Reviewer 2: No, the Applicability is only about being an HTML document.
Block of repeated content is a subjective definition (because of the possibility for two blocks to be "equivalent resources" which is itself subject to interpretation). Therefore, it cannot be used in the Applicability. This in turn shaped the architecture of several rules (and notably the presence of the "document has no block of repeated content" rule.

The Expectations contain a "for each block of repeated content" quantification which is, by definition, true when there is no block of repeated content.

Reviewer 5: So the only applicability for this rule is the HTML document, nevertheless, it has any block of repeated content or not. However, from the description, passed, and failed examples are "block of repeated content" objective.

---
id: 3e12e1
name: Block of repeated content is collapsible
rule_type: atomic
description: |
  This rule checks that repeated blocks of content are collapsible
accessibility_requirements:
  wcag-technique:SCR28: # Using an expandable and collapsible menu to bypass block of content
    forConformance: false
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - Accessibility Tree
  - CSS Styling
  - DOM Tree
acknowledgments:
  authors:
    - Jean-Yves Moyen
  assets:
    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
---

## Applicability

This rule applies to any [HTML web page][].

## Expectation 1

For each [block of repeated content][] in the test target which is before the [main block of content][], there exists a [keyboard actionable][] [instrument][] to make this [block][] not [visible][].

## Expectation 2

> For each [block of repeated content][] in the test target which is before the [main block of content][], there exists an [instrument][], which is [included in the accessibility tree][], to remove this [block][] from the [accessibility tree][included in the accessibility tree].

Reviewer 3: Don't really need the block gone, but its content.

```suggestion
For each [block of repeated content][] in the test target which is before the [main block of content][], there exists an [instrument][], which is [included in the accessibility tree][], to remove all nodes in this [block][] from the [accessibility tree][included in the accessibility tree].
```

- [Technique SCR28: Using an expandable and collapsible menu to bypass block of content][tech scr28]

> In order to focus on only on the part of the associated composite rule ([_Bypass blocks of content_][bypass blocks]) which this atomic rule illustrate, and given the very nature of some of the other input rules, test cases use bold text instead of heading (to avoid also passing rule [_Document has heading for main section of content_][document has heading for main]) and a `<div id="main">` instead of a `main` element (to avoid also passing rule [_Document has a main landmark_][document has main]). These are bad practices and should be avoided. In addition, this often create examples that do not satisfy [Success Criterion 1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships).

Reviewer: Should this be mentioned in the assumptions? Would placing this right underneath the "test cases" header not be more appropriate?

Ditto with all the other such assumptions.

Reviewer 2: These are in the Background section, not in the Assumptions. This is indeed not an Assumption but some high-level explanation valid for all examples.

I do agree that it could also fit under the "Test Cases" headings. Or that we could structure Background a bit better.
We've only started using Background fairly recently (when killing out notes). My current stance is to have first the background about the rule, then the list of links ("bibliography", to some extend), and then the bakground about the test cases. This is likely not optimal…

I'm all up for having a CG discussion on what to do with these extended backgrounds we now have, whether it's sub-headings, move some bit under the Test Cases heading, … I think we start to have enough examples of what we put in Background to start having an idea of what can be done. This is however a separate topic and I suggest keeping it here in this PR and revisiting if we decide to change.

Reviewer: Oh right. Silly GitHub confused me. I am fine with this being in the background. Thank you for the explanation! I will approve for now but if this is moved to the test cases or if backgrounds are restructured better I am fine with that too.

		<a href="#" onclick="toggleHidden('chapters-navigation')">Toggle table of content</a>

		<nav id="chapters-navigation">

> 			<b>Content</b>

Reviewer 3: I would much prefer we take out this element all together. This is a 1.3.1 issue otherwise.

```suggestion
```

			<ol>
				<li><a>Chapter 1</a></li>
				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
			</ol>
		</nav>

		<div id="main">

> 			<b>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</b>

Reviewer 3: Not unreasonable as a paragraph. bold seems odd here.

```suggestion
			<p>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</p>
```
